### PR TITLE
ignore dependencies of build-time-dependency

### PR DESCRIPTION
**Pull request by hyuraku**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----
closes https://github.com/Homebrew/brew/issues/12749

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
I don't think this is the right fix. This should be scoped exclusively to `uninstall` and casks.

File: Library/Homebrew/cask_dependent.rb [hunk 1]
```diff
@@ -17,11 +17,8 @@ def full_name
     @cask.full_name
   end
 
-  def runtime_dependencies(ignore_missing: false)
-    recursive_dependencies(ignore_missing: ignore_missing).reject do |dependency|
-      tags = dependency.tags
-      tags.include?(:build) || tags.include?(:test)
-    end
+  def runtime_dependencies
+    deps.map { |deps| [deps, deps.to_formula.runtime_dependencies] }.flatten.uniq
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cask_dependent.rb:21]:
```suggestion
    deps.flat_map { |dep| [dep, dep.to_formula.runtime_dependencies] }.uniq
```

**Comment by hyuraku** [comment]
`deps.to_formula.runtime_dependencies` should be variable length arguments; otherwise, `runtime_dependencies` would be An array including the other arrays.

**Review by MikeMcQuaid — APPROVED** [review]
Looks good, thanks again @hyuraku!

**Pull request merged**